server/src/core/server/stacks/approveComment.ts

```diff
@@ -64,8 +64,8 @@ const approveComment = async (
   if (
     revision &&
     tenant.integrations.akismet.enabled &&
-    (revision.actionCounts.COMMENT_REPORTED_SPAM > 0 ||
-      revision.actionCounts.COMMENT_DETECTED_SPAM > 0)
+    (revision.actionCounts.FLAG__COMMENT_REPORTED_SPAM > 0 ||
+      revision.actionCounts.FLAG__COMMENT_DETECTED_SPAM > 0)
```

Comment:
the fact that we can accidentally not use the serialized value for this here is rather annoying... everywhere else in the code-base, this is `COMMENT_REPORTED_SPAM` and `COMMENT_DETECTED_SPAM`. You just have to "know" that it's going to do a weird `FLAG__` when serializing and persisting this enum into Mongo.

This speaks to a limitation of using GraphQL enum types inside your data persistence instead of creating concrete maps and persistence in a data layer. It's important to know you're accurately saving, retrieving, and comparing data types.

Doubly annoying is, this `actionCounts` object is not typed in any way. It is a `Record<string, number>` and has no indication of what any of the counts are indexed as. It's all just hidden "magic" key values and there is no way to know this by reading the code. You have to debug this and realize "oh wow, the enum serialization does not match the names of the types/values we use everywhere else".

Frustrating.